src/drivers/virtual_timer_controller.cpp

```diff
@@ -0,0 +1,136 @@
+#include "drivers/virtual_timer_controller.hpp"
+
+#include "drivers/virtual_timer_linked_list.hpp"
+#include "nrf52833.h"
+#include "scheduler/pending_process_callbacks.hpp"
+
+namespace edge::drivers {
+
+extern "C" {
+void TIMER3_IRQHandler(void)
+{
+    VirtualTimerController vtc = VirtualTimerController::get();
```

Comment:
1. this is wrong
2. you are only able to do this because you forgot to delete the copy assignment operator

Reply:
lmao